### PR TITLE
benchmarks for sqlite storage

### PR DESCRIPTION
**Pull request by eugene-babichenko**

### CODE REVIEW

**Comment by eugene-babichenko** [comment]
BTW those are results on my machine (change is in comparison to using in-memory index from the new storage module):

```
put_block               time:   [241.33 us 255.75 us 272.66 us]                       
                        change: [+99.724% +110.41% +121.45%]

get_block               time:   [14.755 us 14.776 us 14.802 us]                       
                        change: [+30.425% +32.269% +33.734%] 
```

File: chain-storage-sqlite-old/benches/storage.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#[macro_use]
+extern crate criterion;
+extern crate tempfile;
```
Comment (NicolasDP) [inline chain-storage-sqlite-old/benches/storage.rs]: do you still need to include `extern crate` ? This project is not edition 2018?

File: chain-storage-sqlite-old/Cargo.toml [hunk 2]
```diff
@@ -21,5 +21,11 @@ version = "0.16.0"
 features = ["bundled"]
 
 [dev-dependencies]
+criterion = "0.3.0"
```
Comment (NicolasDP) [inline chain-storage-sqlite-old/Cargo.toml]: can these new dependencies be included only when `cargo bench` is launch? Just adding less deps that are not use when running the bench?

File: chain-storage-sqlite-old/benches/storage.rs (hunk 1 above)
Comment (eugene-babichenko) [reply chain-storage-sqlite-old/benches/storage.rs]: Thanks for noting this, it doesn't really make sense here.

File: chain-storage-sqlite-old/Cargo.toml (hunk 2 above)
Comment (eugene-babichenko) [reply chain-storage-sqlite-old/Cargo.toml]: Hm, that's a bit tricky with the current setup. The working solution would be to:

- move `bench`-only dependencies
- mark them optional
- join them under a feature with the name like `with-bench`
- require this feature for `bench` builds

This *should* be fine, but requires activating features with `cargo` flags which don't really work with cargo workspaces right now.

Comment (eugene-babichenko) [reply chain-storage-sqlite-old/Cargo.toml]: Apparently, running that command from within `chain-storage-sqlite-old` works just fine. The workspace is still recognized correctly and the top-level `target` directory is used. That's just a bit inconvenient.

**Comment by vincenthz** [comment]
do you have the number for the in-memory `put_block` and `get_block` ? I have trouble parsing the change properly.

**Pull request merged**